src/main/java/core/basesyntax/JavaStreamApi.java

```diff
@@ -48,7 +58,11 @@ public String findElement(List<String> elements, String element) {
      * NoSuchElementException</p>
      **/
     public Double averageSumOdd(List<Integer> numbers) {
-        return null;
+        return IntStream.range(0, numbers.size())
+                .map(x -> x % 2 == 1 ? numbers.get(x) - 1 : numbers.get(x))
+                .filter(x -> x % 2 == 1)
+                .average()
+                .orElseThrow(NoSuchElementException::new);
```

Comment:
take a look to orElseThrow() without parameters

Reply:
I do not understand this remark

Reply:
read about this method
which exception is thrown by default?

Reply:
Done!